pkg/cloud/aliyunprovider.go

```diff
@@ -0,0 +1,828 @@
+}
+
+type AliyunPVKey struct {
+	ProviderID       string
+	RegionID         string
+	DiskType         string
+	DiskCategory     string
+	PerformaceLevel  string
+	StorageClassName string
+}
+
+func (aliyunPVKey *AliyunPVKey) Features() string {
+	keyLookup := stringutil.DeleteEmptyStringsFromArray([]string{aliyunPVKey.RegionID, aliyunPVKey.DiskType, aliyunPVKey.DiskCategory, aliyunPVKey.PerformaceLevel})
+	return strings.Join(keyLookup, "::")
+}
+
+func (aliyunPVKey *AliyunPVKey) ID() string {
+	return aliyunPVKey.ProviderID
+}
+
+// Get storage class information for PV.
+func (aliyunPVKey *AliyunPVKey) GetStorageClass() string {
+	return aliyunPVKey.StorageClassName
+}
+
+// Helper functions for aliyunprovider.go
+
+// createDescribePriceACSRequest creates the HTTP GET request for the required resources' Price information,
+// When supporting subscription and Premptible resources this HTTP call needs to be modified with PriceUnit information
+// When supporting different new type of instances like Compute Optimized, Memory Optimized etc make sure you add the instance type
+// in unit test and check if it works or not to create the ack request and processDescribePriceAndCreateAliyunPricing function
+// else more paramters need to be pulled from kubernetes node response or gather infromation from elsewhere and function modified.
+// TO-DO: Add disk adjustments to the node , Test it out!
+func createDescribePriceACSRequest(i interface{}) (*requests.CommonRequest, error) {
+	request := requests.NewCommonRequest()
+	request.Method = requests.GET
+	request.Product = ALIYUN_ECS_PRODUCT_CODE
+	request.Domain = ALIYUN_ECS_DOMAIN
+	request.Version = ALIYUN_ECS_VERSION
+	request.Scheme = requests.HTTPS
+	request.ApiName = ALIYUN_DESCRIBE_PRICE_API_ACTION
+	switch i.(type) {
+	case *SlimK8sNode:
+		node := i.(*SlimK8sNode)
+		request.QueryParams["RegionId"] = node.RegionID
+		request.QueryParams["ResourceType"] = ALIYUN_INSTANCE_RESOURCE_TYPE
+		request.QueryParams["InstanceType"] = node.InstanceType
+		request.QueryParams["PriceUnit"] = node.PriceUnit
+		// For Enhanced General Purpose Type g6e SystemDisk.Category param doesn't default right,
+		// need it to be specifically assigned to "cloud_ssd" otherwise there's errors
+		if node.InstanceTypeFamily == ALIYUN_ENHANCED_GENERAL_PURPOSE_TYPE {
+			request.QueryParams["SystemDisk.Category"] = ALIYUN_SYSTEMDISK_CLOUD_ESSD_CATEGORY
+		}
+		request.TransToAcsRequest()
+		return request, nil
+	case *SlimK8sDisk:
+		disk := i.(*SlimK8sDisk)
+		request.QueryParams["RegionId"] = disk.RegionID
+		request.QueryParams["ResourceType"] = ALIYUN_DISK_RESOURCE_TYPE
+		request.QueryParams["DataDisk.1.Category"] = disk.DiskCategory
+		request.QueryParams["DataDisk.1.Size"] = fmt.Sprintf("%d", disk.SizeInGiB)
+		request.QueryParams["PriceUnit"] = disk.PriceUnit
+		request.TransToAcsRequest()
+		return request, nil
+	default:
+		return nil, fmt.Errorf("unsupported ECS type for DescribePrice at this time")
```

Comment:
You can add the type to help the user:
```suggestion
	default:
		return nil, fmt.Errorf("unsupported ECS type (%T) for DescribePrice at this time", i)
```